tests/platform_tests/api/test_fan.py

```diff
@@ -0,0 +1,208 @@
+        # Ensure the fan speed is sane
+        for i in range(self.num_fans):
+            speed = fan.get_speed(platform_api_conn, i)
+            logger.error("vaibhav got fan speed {}".format(speed))
+            if self.expect(speed is not None, "Unable to retrieve Fan {} speed".format(i)):
+                if self.expect(isinstance(speed, int), "Fan {} speed appears incorrect".format(i)):
+                    self.expect(speed > 0 and speed <= 100, "Fan {} speed  {} reading is not within range".format(i, speed))
+        self.assert_expectations()
+
+    def test_get_direction(self, duthost, localhost, platform_api_conn):
+        # Ensure the fan speed is sane
+        FAN_DIRECTION_LIST = [
+            "intake",
+            "exhaust",
+            "N/A",
+        ]
+        for i in range(self.num_fans):
+            direction = fan.get_direction(platform_api_conn, i)
+            if self.expect(direction is not None, "Unable to retrieve Fan {} direction".format(i)):
+                self.expect(direction in FAN_DIRECTION_LIST,"Fan {} direction is not one of predefined directions".format(i))
+
+        self.assert_expectations()
+    
+    def test_get_fans_speed_target(self, duthost, localhost, platform_api_conn):
+        try:
+            num_fans = int(chassis.get_num_fans(platform_api_conn))
+        except:
+            pytest.fail("num_fans is not an integer")
+
+        fan_list = chassis.get_all_fans(platform_api_conn)
+        pytest_assert(fan_list is not None, "Failed to retrieve fans")
+
+        for i in range(num_fans):
+            fan_instance = chassis.get_fan(platform_api_conn, i)
+            self.expect(fan_instance and fan_instance == fan_list[i], "Fan {} is incorrect".format(i))
+            speed_target_val = 25
+            speed_set= fan.set_speed(platform_api_conn, i, speed_target_val)
+            target_speed = fan.get_target_speed(platform_api_conn, i)
+            if self.expect(target_speed is not None, "Unable to retrieve Fan {} target speed".format(i)):
+                if self.expect(isinstance(target_speed, int), "Fan {} target speed appears incorrect".format(i)):
+                    self.expect(target_speed == speed_target_val, "Fan {} target speed setting is not correct, speed_target_val {} target_speed = {}".format(i, speed_target_val, target_speed))
+
+        self.assert_expectations()
+
+    def test_get_fans_speed_tolerance(self, duthost, localhost, platform_api_conn):
+        try:
+            num_fans = int(chassis.get_num_fans(platform_api_conn))
+        except:
+            pytest.fail("num_fans is not an integer")
+
+        fan_list = chassis.get_all_fans(platform_api_conn)
+        pytest_assert(fan_list is not None, "Failed to retrieve fans")
+
+        for i in range(num_fans):
+            fan_instance = chassis.get_fan(platform_api_conn, i)
+            self.expect(fan_instance and fan_instance == fan_list[i], "Fan {} is incorrect".format(i))
```

Comment:
As above, getting the fan list and comparing against the result of `get_fan(index)` is not necessary here, as this consistency check is already covered by the `test_fans()` function in test_chassis.py.

Reply:
fixed